meilisearch-http/src/index/search.rs

```diff
@@ -210,15 +225,61 @@ impl Index {
     }
 }
 
+fn compute_matches<A: AsRef<[u8]>>(
+    matcher: &impl Matcher,
+    document: &Document,
+    analyzer: &Analyzer<A>
+    ) -> MatchesInfo {
+    let mut matches = BTreeMap::new();
+
+    for (key, value) in document {
+        let mut infos = Vec::new();
+        compute_value_matches(&mut infos, value, matcher, &analyzer);
+        if !infos.is_empty() {
+            matches.insert(key.clone(), infos);
+        }
+    }
+    matches
+}
+
+fn compute_value_matches<'a, A: AsRef<[u8]>>(
+    infos: &mut Vec<MatchInfo>,
+    value: &Value,
+    matcher: &impl Matcher,
+    analyzer: &Analyzer<'a, A>,
+) {
+    match value {
+        Value::String(s) => {
+            let analyzed = analyzer.analyze(s);
+            let mut start = 0;
+            for (word, token) in analyzed.reconstruct() {
+                if token.is_word() {
+                    if let Some(length) = matcher.matches(token.text()) {
+                        infos.push(MatchInfo { start, length });
+                    }
+                }
+
+                start += word.len();
+            }
+        }
+        Value::Array(vals) => vals
+            .iter()
+            .for_each(|val| compute_value_matches(infos, val, matcher, analyzer)),
+        Value::Object(vals) => vals
+            .values()
+            .for_each(|val| compute_value_matches(infos, val, matcher, analyzer)),
```

Comment:
I am not sure `matchesInfos` were computed on other things than strings in MeiliSearch.

Reply:
Oh yes I missed this! matches is not compute on arrays and objects!